### PR TITLE
VACMS-3792: JSD in CMS + help center updates

### PR DESCRIPTION
**Pull request by oksana-c**
## Description

See #3792 , #2731

## Testing done

Tested with real users in a research phase.

## QA steps

- [ ] JSD Widget is rendered on all CMS Help Center pages with an exception of CMS Help Center Homepage (node 6454, path / help)
- [ ]  JSD Widget is rendered on /user/login page and on front page `/` when viewed as anonymous user
- [ ] Admin toolbar menu changes
     - [ ] top level CMS help center link "? Help" should open in the same tab when clicked and have no submenus
- [ ] "helpdesk" link on the login page under the password field leads to the JSD portal instead of launching email client
- [ ] Edit /help node. Add a `<p id="test">Test</p> and <a href="/help" target="_blank">Help opens in new tab</a>`  in a wysiwyg field, using the source view mode. Save and verify that both `id` and `target` attributes are preserved .

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [x] `Core Application Team`
- [ ] `Product Support Team`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [x] Yes, and it's written in issue #3815  and queued for publication. 
  - [x] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by oksana-c** [comment]
@ndouglas I'd be very much obliged if you could take a look at this test and see what I may be missing to get the widget recognized on the page - https://github.com/department-of-veterans-affairs/va.gov-cms/pull/3850/commits/e9cef94bf17a2bfcc750af753eadfb3f526556ea 

tried adding wait to see maybe it's a JS loading issues, but no matter how long the wait is - the widget markup is still not on the page. this may be an issue with the browser we use for tests, and if it is, we'll have to give up this test temporarily. what do you think?

goal: test presense of the JSD widget under anonymous and logged in user
Screen Shot 2020-12-22 at 3 14 59 PM

**Comment by ndouglas** [comment]
I would bet it's because of the Goutte driver, or at least it seems like (testing locally) the issues I've had with other tests where I was trying to test JS behavior.  I tried using the `@javascript` tag to trigger using PhantomJS, and changed a few things from there but it starts complaining about not being able to find XPaths because there is some further incompatibility or shortcoming of PhantomJS.

I'm happy to dig into this further tomorrow if you want, but probably the fastest approach is just what you said 😞

**Comment by oksana-c** [comment]
@ndouglas , gotcha. Yes, agreed on skipping tests for this one. Manual QA should suffice.

**Comment by oksana-c** [comment]
⚠️ 🛑  **TO ALL WHO SEE THIS PR - PLEASE DO NOT UPDATE THIS PR**. 
This PR environment will be used to conduct user testing and contains content that should not be erased by a rebuild. 
Thank you!

File: tests/behat/Drupal/FeatureContext.php [hunk 1]
```diff
@@ -55,6 +55,18 @@ public function cleanUp() {
     }
   }
 
+  /**
```
Comment (oksana-c) [inline tests/behat/Drupal/FeatureContext.php:58]: reminder to myself to remove this before merging

File: tests/behat/Drupal/FeatureContext.php [hunk 2]
```diff
@@ -55,6 +55,18 @@ public function cleanUp() {
...
+   * Wait a while, for JS loading or debugging.
+   *
+   * @param int $seconds
+   *   How long to wait.
+   *
+   * @When I wait :seconds second(s)
+   */
+  public function wait($seconds) {
```
Comment (oksana-c) [inline tests/behat/Drupal/FeatureContext.php:66]: reminder to myself to remove this before merging

**Comment by oksana-c** [comment]
⚠️ 🛑 PLEASE DO NOT UPDATE OR MERGE ANYTHING TO THIS PR.
This PR environment will be used to conduct user testing and contains content that should not be erased by a rebuild.
Thank you!

**Comment by indytechcook** [comment]
@oksana-c build failed: db is out of sync with code.  Master needs to be merged in:

```
2020-12-31T23:08:19.022Z - 5fe26e532c4f12e655455600# /bin/sh -c drush config:import -y
 C  Config                                                            Operation                
 o                                                                                             
 l                                                                                             
 l                                                                                             
 e                                                                                             
 c                                                                                             
 t                                                                                             
 i                                                                                             
 o                                                                                             
 n                                                                                             
    core.extension                                                    update 
    core.menu.static_menu_link_overrides                              update 
    field.field.node.documentation_page.field_content_block           update 
    filter.format.rich_text                                           update 
    editor.editor.rich_text                                           update 
    views.view.right_sidebar_latest_revision                          update 
    toolbar_menu.toolbar_menu_element.cms_help_center                 delete   
    views.view.advancedqueue_jobs                                     delete   
    core.entity_view_display.media.video.user_guides                  delete   
    core.entity_view_display.paragraph.downloadable_file.user_guides  delete   
    core.entity_view_display.paragraph.media_list_videos.user_guides  delete   
    core.entity_view_mode.media.user_guides                           delete   
    advancedqueue.advancedqueue_queue.command_runner                  delete
Import the listed configuration changes? (y/n): y
Drupal\Core\Extension\Exception\UnknownExtensionException: The module[error]
advancedqueue does not exist. in
/var/lib/tugboat/docroot/core/lib/Drupal/Core/Extension/ExtensionList.php:522
Stack trace:
#0
/var/lib/tugboat/docroot/core/lib/Drupal/Core/Extension/ExtensionList.php(541):
Drupal\Core\Extension\ExtensionList->getPathname('advancedqueue')
#1
/var/lib/tugboat/docroot/core/lib/Drupal/Core/Extension/ModuleInstaller.php(189):
Drupal\Core\Extension\ExtensionList->getPath('advancedqueue')
#2
/var/lib/tugboat/docroot/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php(83):
Drupal\Core\Extension\ModuleInstaller->install(Array, false)
#3
/var/lib/tugboat/docroot/core/lib/Drupal/Core/Config/ConfigImporter.php(813):
Drupal\Core\ProxyClass\Extension\ModuleInstaller->install(Array,
false)
#4
/var/lib/tugboat/docroot/core/lib/Drupal/Core/Config/ConfigImporter.php(577):
Drupal\Core\Config\ConfigImporter->processExtension('module',
'install', 'va_gov_help_cen...')
#5
/var/lib/tugboat/docroot/core/lib/Drupal/Core/Config/ConfigImporter.php(514):
Drupal\Core\Config\ConfigImporter->processExtensions(Array)
#6
/var/lib/tugboat/docroot/vendor/drush/drush/commands/core/config.drush.inc(688):
Drupal\Core\Config\ConfigImporter->doSyncStep('processExtensio...',
Array)
#7
/var/lib/tugboat/docroot/vendor/drush/drush/includes/drush.inc(727):
_drush_config_import(Object(Drupal\Core\Config\StorageComparer))
#8
/var/lib/tugboat/docroot/vendor/drush/drush/includes/drush.inc(713):
drush_call_user_func_array('_drush_config_i...', Array)
#9
/var/lib/tugboat/docroot/vendor/drush/drush/commands/core/config.drush.inc(669):
drush_op('_drush_config_i...',
Object(Drupal\Core\Config\StorageComparer))
#10
/var/lib/tugboat/docroot/vendor/drush/drush/includes/command.inc(422):
drush_config_import()
#11
/var/lib/tugboat/docroot/vendor/drush/drush/includes/command.inc(231):
_drush_invoke_hooks(Array, Array)
#12
/var/lib/tugboat/docroot/vendor/drush/drush/includes/command.inc(199):
drush_command()
#13
/var/lib/tugboat/docroot/vendor/drush/drush/lib/Drush/Boot/BaseBoot.php(67):
drush_dispatch(Array)
#14
/var/lib/tugboat/docroot/vendor/drush/drush/includes/preflight.inc(67):
Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#15 phar:///usr/local/bin/drush/bin/drush.php(153): drush_main()
#16 /usr/loc
```

image


If tugbaot has an error in github this mostly likey our build failed and you should look at logs to see why

**Comment by oksana-c** [comment]
> @oksana-c build failed: db is out of sync with code. Master needs to be merged in:

@indytechcook  this is interesting and not the behavior that I would expect on a PR environment.

I've some questions to clarify:

1. Does Tugboat regularly rebuild any PR environment even if it is inactive?
2. If (1) is yes, why does Tugboat try to pull a new DB? My understanding (and it may be off a bit) is that when PR builds - the DB remains the same until the PR is closed based on the initial image built for that PR.

**Comment by oksana-c** [comment]
@jordanamello - new test environment link https://pr3850-geb9i1vxs7ugc9kzpbajhiyfirxs1npl.ci.cms.va.gov/ with latest help center homepage published,

**Comment by acrollet** [comment]
@oksana-c FYI there's a ['lock' option](https://docs.tugboat.qa/building-a-preview/administer-previews/change-preview-states/#lock-unlock) on tugboat environments that should prevent them from being rebuilt.

**Comment by oksana-c** [comment]
super useful! Thanks @acrollet. Locked this one 👍

**Comment by oksana-c** [comment]
@acrollet @ndouglas or @indytechcook , would be good to have one of you to look thru this one before I merge. Nothing heavy here - simple sanity check on ACs and code.

**Comment by ndouglas** [comment]
I'll be happy to look at it tomorrow morning, unless someone else gets to it first 😄

File: docroot/modules/custom/va_gov_login/va_gov_login.module [hunk 3]
```diff
@@ -13,7 +13,10 @@ use Drupal\Component\Render\FormattableMarkup;
  */
 function va_gov_login_form_alter(&$form, &$form_state, $form_id) {
   if ($form_id == "user_login_form") {
-    $form['pass']["#description"] = FieldFilteredMarkup::create(t('Enter the password that accompanies your username, if you have an issue with your password please email the @helpdesk.', ['@helpdesk' => new FormattableMarkup('<a href="mailto:vacmssupport@va.gov">helpdesk</a>', [])]));
+    $form['pass']["#description"] = FieldFilteredMarkup::create(t('Enter the password that accompanies your username, if you have an issue with your password please contact @helpdesk.',
```
Comment (acrollet) [inline docroot/modules/custom/va_gov_login/va_gov_login.module:16]:
```suggestion
    $form['pass']['#description'] = FieldFilteredMarkup::create(t('Enter the password that accompanies your username. If you have an issue with your password please contact @helpdesk.',
```

File: docroot/modules/custom/va_gov_help_center/va_gov_help_center.module [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+function va_gov_help_center_toolbar() {
+  $items = [];
+
+  // Adds a link to CMS help center to the top level of admin toolbar.
+  $items['help-center'] = [
+    '#type' => 'toolbar_item',
+    '#weight' => 1,
+    'tab' => [
+      '#type' => 'link',
+      '#url' => Url::fromRoute('entity.node.canonical', ['node' => 6454]),
```
Comment (acrollet) [inline docroot/modules/custom/va_gov_help_center/va_gov_help_center.module:23]: Hard-coded NIDs make me sad, but I can't think of a solution that's not a lot more effort...

**Review by acrollet — CHANGES_REQUESTED** [review]
A couple minor suggestions, I'm seeing that there are a couple new accessibility issues on the login page.

File: docroot/modules/custom/va_gov_help_center/va_gov_help_center.module (hunk 4 above)
Comment (oksana-c) [reply docroot/modules/custom/va_gov_help_center/va_gov_help_center.module:23]: since URL can change -this is the most stable option for now

**Comment by oksana-c** [comment]
apparently [Attlasian is not planning on fixing accessibility issues with their widget](https://community.atlassian.com/t5/Jira-Service-Management/Jira-Help-Desk-Widget-contains-an-empty-iframe-and-will-not-pass/qaq-p/850377), so I excluded it for homepage scan.

**Review by acrollet — APPROVED** [review]

**Pull request merged**